src/Axiom/Set/Rel.agda

```diff
@@ -88,6 +90,54 @@ dom∈ {R = R} {a} =
   (∃[ b ] (a , b) ∈ R)            ∎
   where open R.EquationalReasoning
 
+module _ {x : A} {y : B} where
+  module _ {a : A} where
+    ∈-dom-singleton-pair : a ≡ x ⇔ a ∈ dom ❴ x , y ❵
+    ∈-dom-singleton-pair = mk⇔ (λ a≡x → from dom∈ (y , to ∈-singleton (×-≡,≡→≡ (a≡x , refl))))
+                               (,-injectiveˡ ∘ from ∈-singleton ∘ proj₂ ∘ to dom∈)
+
+    dom-single→single : a ∈ dom ❴ x , y ❵ → a ∈ ❴ x ❵
+    dom-single→single = to ∈-singleton ∘ from ∈-dom-singleton-pair
+
+    single→dom-single : a ∈ ❴ x ❵ → a ∈ dom ❴ x , y ❵
+    single→dom-single = to ∈-dom-singleton-pair ∘ from ∈-singleton
+
+  dom-single≡single : dom ❴ x , y ❵ ≡ᵉ ❴ x ❵
+  dom-single≡single = dom-single→single , single→dom-single
+
+∈-dom : {a : A × B} → a ∈ R → proj₁ a ∈ dom R
+∈-dom {a = a} a∈ = to ∈-map (a , (refl , a∈))
+
+∉-dom∅ : {a : A} → a ∉ dom{A}{B} ∅
+∉-dom∅ {a} a∈dom∅ = ⊥-elim $ ∉-∅ $ proj₂ $ (to dom∈) a∈dom∅
+
+dom∅ : dom{A}{B} ∅ ≡ᵉ ∅
+dom∅ = ⊥-elim ∘ ∉-dom∅ , ∅-minimum (dom ∅)
+
+dom∪ : dom (R ∪ R') ≡ᵉ (dom R ∪ dom R')
+dom∪ = from ≡ᵉ⇔≡ᵉ' dom∪'
+  where
+  dom∪' : (dom (R ∪ R')) ≡ᵉ' (dom R ∪ dom R')
+  dom∪' {R = R} {R'} a = let open R.EquationalReasoning in
+    a ∈ dom (R ∪ R')                           ∼⟨ dom∈ ⟩
+    (∃[ b ] (a , b) ∈ R ∪ R')                  ∼⟨ ∃-cong′ (R.SK-sym ∈-∪) ⟩
+    (∃[ b ] ((a , b) ∈ R ⊎ (a , b) ∈ R'))      ∼⟨ mk⇔ (λ {(b , inj₁ p) → inj₁ (b , p);
+                                                          (b , inj₂ p) → inj₂ (b , p)})
+                                                      (λ {(inj₁ (b , pf)) → (b , inj₁ pf);
+                                                          (inj₂ (b , pf)) → (b , inj₂ pf)}) ⟩
+    (∃[ b ] (a , b) ∈ R ⊎ ∃[ b ] (a , b) ∈ R') ∼⟨ mk⇔ (λ {(inj₁ bpf) → inj₁ (from dom∈ bpf);
+                                                          (inj₂ bpf) → inj₂ (from dom∈ bpf)})
+                                                      (λ {(inj₁ pf) → inj₁ (to dom∈ pf);
+                                                          (inj₂ pf) → inj₂ (to dom∈ pf)}) ⟩
```

Comment:
You can use `∃-cong′` to simplify the rest of these steps as well.

Reply:
I tried to do so last week, but couldn't get anything simpler than what I had to type check... but I'll try again.

Reply:
Here you go:
```
    (∃[ b ] ((a , b) ∈ R ⊎ (a , b) ∈ R'))      ↔⟨ ∃-distrib-⊎' ⟩
    (∃[ b ] (a , b) ∈ R ⊎ ∃[ b ] (a , b) ∈ R') ∼⟨ R.SK-sym (dom∈ ⊎-cong dom∈) ⟩
```
Only needs the correct imports